Modules/_testcapimodule.c

```diff
@@ -4776,6 +4798,7 @@ static PyMethodDef TestMethods[] = {
     {"get_mapping_items", get_mapping_items, METH_O},
     {"test_pythread_tss_key_state", test_pythread_tss_key_state, METH_VARARGS},
     {"hamt", new_hamt, METH_NOARGS},
+    {"bad_get", bad_get, METH_FASTCALL},
```

Comment:
Is there a particular reason to use `METH_FASTCALL` here? Can't we just use `METH_VARARGS`? Did you use it because it was easy to reproduce the crash in Python 3?

Reply:
Also, if it's not possible to crash the interpreter in a pure Python 3 code, I wonder whether we should only fix this in Python 2.7.

Reply:
> if it's not possible to crash the interpreter in a pure Python 3 code

I never said that it's impossible. I just said that I didn't manage to. There is a lot of C code in CPython, I didn't audit all of it to ensure that it can't be used to get this crash. Even if I did, somebody might add a function tomorrow which does allow to induce the crash.

Reply:
> Is there a particular reason to use METH_FASTCALL here?

Absolutely. When calling with `METH_VARARGS`, a temporary tuple is constructed which holds an additional reference. So there won't be a crash in that case.